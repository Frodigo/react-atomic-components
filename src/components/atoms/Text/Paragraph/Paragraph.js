import React from 'react';
import PropTypes from 'prop-types';

/**
 * Paragraph wrapper.
 * Please add `inverse` class if you want to show heading with white color.
 */
const Paragraph = (props) => {
    const { children, type, classes, textAlign } = props;

    let parahraph = null;

    const styles = {
        textAlign
    };

    const customClasses = classes ? classes : '';

    switch (type) {
        case 'normal': {
            if (customClasses) {
                parahraph = (
                    <p className={classes} style={styles}>
                        {children}
                    </p>
                );

                break;
            }

            parahraph = (
                <p className="paraghaph" style={styles}>
                    {children}
                </p>
            );

            break;
        }

        case 'smaller': {
            parahraph = (
                <p
                    className={`paraghaph paragraphSmaller ${classes}`}
                    style={styles}
                >
                    {children}
                </p>
            );

            break;
        }

        case 'large': {
            parahraph = (
                <p
                    className={`paraghaph paragraphLarge ${classes}`}
                    style={styles}
                >
                    {children}
                </p>
            );

            break;
        }

        default: {
            parahraph = <p>{children}</p>;

            break;
        }
    }

    return <>{parahraph}</>;
};

Paragraph.propTypes = {
    children: PropTypes.string,
    classes: PropTypes.string,
    textAlign: PropTypes.string,
    type: PropTypes.string.isRequired
};

Paragraph.defaultProps = {
    type: 'normal',
    classes: ''
};

export default Paragraph;
