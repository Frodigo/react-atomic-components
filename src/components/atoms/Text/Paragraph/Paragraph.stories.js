import React from 'react';

import Paragraph from './Paragraph';

export default {
    title: 'Atoms/Paragraphs',
    component: Paragraph,
    argTypes: {
        type: {
            control: {
                type: 'select',
                options: ['normal', 'large', 'smaller']
            }
        },
        textAlign: {
            control: {
                type: 'select',
                options: ['left', 'center', 'right']
            }
        }
    }
};

const Template = (args) => <Paragraph {...args} />;

export const ParagraphDefault = Template.bind({});
ParagraphDefault.args = {
    children:
        'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie. '
};

const InverseTemplate = (args) => (
    <div
        style={{
            backgroundColor: 'black',
            padding: `${30}px`
        }}
    >
        <Paragraph {...args} />
    </div>
);

export const ParagraphInverse = InverseTemplate.bind({});
ParagraphInverse.args = {
    children:
        'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie. ',
    classes: 'inverse'
};
