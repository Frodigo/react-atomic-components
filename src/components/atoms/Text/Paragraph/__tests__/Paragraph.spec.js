import React from 'react';
import { render } from '@testing-library/react';

import Paragraph from '../Paragraph';

describe('Paragraph should be rendered', () => {
    test('Paragraph renders correctly', () => {
        const { container } = render(<Paragraph>Paragraph</Paragraph>);

        expect(container).toMatchSnapshot();
    });

    test('Paragraph large renders correctly', () => {
        const { container } = render(
            <Paragraph type={'large'}>Paragraph</Paragraph>
        );

        expect(container).toMatchSnapshot();
    });

    test('Paragraph smaller renders correctly', () => {
        const { container } = render(
            <Paragraph type={'smaller'}>Paragraph</Paragraph>
        );

        expect(container).toMatchSnapshot();
    });

    test('Paragraph inverse should render correctly', () => {
        const { container } = render(
            <Paragraph classes="inverse">Paragraph inverse</Paragraph>
        );

        expect(container).toMatchSnapshot();
        expect(container.firstChild.classList.contains('inverse')).toBe(true);
    });

    test('Paragraph with specific text align should render correctly', () => {
        const { container } = render(
            <Paragraph textAlign="center">Paragraph centered</Paragraph>
        );

        expect(container).toMatchSnapshot();
        expect(container.firstChild.style.textAlign).toBe('center');
    });
});
