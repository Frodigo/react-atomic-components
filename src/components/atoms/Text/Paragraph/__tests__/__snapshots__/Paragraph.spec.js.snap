// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should be rendered Paragraph inverse should render correctly 1`] = `
<div>
  <p
    class="inverse"
  >
    Paragraph inverse
  </p>
</div>
`;

exports[`Paragraph should be rendered Paragraph large renders correctly 1`] = `
<div>
  <p
    class="paraghaph paragraphLarge "
  >
    Paragraph
  </p>
</div>
`;

exports[`Paragraph should be rendered Paragraph renders correctly 1`] = `
<div>
  <p
    class="paraghaph"
  >
    Paragraph
  </p>
</div>
`;

exports[`Paragraph should be rendered Paragraph smaller renders correctly 1`] = `
<div>
  <p
    class="paraghaph paragraphSmaller "
  >
    Paragraph
  </p>
</div>
`;

exports[`Paragraph should be rendered Paragraph with specific text align should render correctly 1`] = `
<div>
  <p
    class="paraghaph"
    style="text-align: center;"
  >
    Paragraph centered
  </p>
</div>
`;
